Parsers/ASimNetworkSession/Parsers/vimNetworkSessionFortinetFortiGate.yaml

```diff
@@ -0,0 +1,284 @@
+    , "92","MTP"
+    , "93","AX.25"
+    , "94","IPIP"
+    , "95","MICP (deprecated)"
+    , "96","SCC-SP"
+    , "97","ETHERIP"
+    , "98","ENCAP"
+    , "99",""
+    , "100","GMTP"
+    , "101","IFMP"
+    , "102","PNNI"
+    , "103","PIM"
+    , "104","ARIS"
+    , "105","SCPS"
+    , "106","QNX"
+    , "107","A/N"
+    , "108","IPComp"
+    , "109","SNP"
+    , "110","Compaq-Peer"
+    , "111","IPX-in-IP"
+    , "112","VRRP"
+    , "113","PGM"
+    , "114",""
+    , "115","L2TP"
+    , "116","DDX"
+    , "117","IATP"
+    , "118","STP"
+    , "119","SRP"
+    , "120","UTI"
+    , "121","SMP"
+    , "122","SM (deprecated)"
+    , "123","PTP"
+    , "124","ISIS over IPv4"
+    , "125","FIRE"
+    , "126","CRTP"
+    , "127","CRUDP"
+    , "128","SSCOPMCE"
+    , "129","IPLT"
+    , "130","SPS"
+    , "131","PIPE"
+    , "132","SCTP"
+    , "133","FC"
+    , "134","RSVP-E2E-IGNORE"
+    , "135","Mobility Header"
+    , "136","UDPLite"
+    , "137","MPLS-in-IP"
+    , "138","manet"
+    , "139","HIP"
+    , "140","Shim6"
+    , "141","WESP"
+    , "142","ROHC"
+    , "143","Ethernet"
+    , "253",""
+    , "254",""
+    , "255","Reserved"];
+  let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let Parser=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]), dstportnumber:int=int(null), hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]), eventresult:string='*', disabled:bool=false){
+    CommonSecurityLog
+    | where not(disabled)
+    | where (isnull(starttime) or TimeGenerated>=starttime) and (isnull(endtime) or TimeGenerated<=endtime)
+    | where DeviceVendor == "Fortinet" and DeviceProduct startswith "FortiGate" and AdditionalExtensions has "cat=traffic"
+    | project Activity,AdditionalExtensions,DestinationIP,DestinationPort,DeviceAction,DeviceInboundInterface,DeviceName,DeviceOutboundInterface,DeviceProduct,DeviceVersion,LogSeverity,Protocol,ReceivedBytes,SentBytes,SourceIP,SourcePort,TimeGenerated
+    | where DeviceAction != "dns"
+    | where  (isnull(dstportnumber) or DestinationPort==dstportnumber) and (array_length(hostname_has_any)==0)
+    | project-rename DstBytes = ReceivedBytes
+      , DstInterfaceName = DeviceOutboundInterface
+      , DstIpAddr = DestinationIP
+      , DstPortNumber = DestinationPort
+      , Dvc = DeviceName
+      , EventEndTime = TimeGenerated
+      , EventMessage = Activity
+      , EventOriginalSeverity = LogSeverity
+      , EventProduct = DeviceProduct
+      , EventProductVersion = DeviceVersion
+      , SrcBytes = SentBytes
+      , SrcInterfaceName = DeviceInboundInterface
+      , SrcIpAddr = SourceIP
+      , SrcPortNumber = SourcePort
+    | extend temp_SrcMatch=has_any_ipv4_prefix(SrcIpAddr,src_or_any)

```

Comment:
It may not change much in this case, but orthodoxy is that you filter prior to the rest, so these lines should go before the project-rename (and work on the original fields).